src/InstrumentationEngine.ProfilerProxy/dllmain.cpp

```diff
@@ -327,16 +327,14 @@ namespace ProfilerProxy
             }
         }
 #endif
+        // Guard against g_hProfiler if it's currently being loaded.
+        CCriticalSectionHolder lock(&g_criticalSection);
 
-        // initializes CProxyLogging only once; shutdown only occurs on DllCanUnloadNow()
-        CProxyLogging::Initialize();
+        CProxyLoggingHolder proxyLoggingHolder;
```

Comment:
>CProxyLoggingHolder proxyLoggingHolder; [](start = 8, length = 39)

Looking at the code, if the proxy logger is going to only be used in this function, does it really need to be global? Would this work if you just got rid of global logger, and created a local EventLogger, keeping the shutdown in the EventLogger's destructor?

Reply:
Yeah, that would be a better implementation and avoid a bunch of headaches

---
In reply to: [365923973](https://github.com/microsoft/CLRInstrumentationEngine/pull/191#discussion_r365923973) [](ancestors = 365923973)